gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -730,6 +726,11 @@ public SeekableByteChannel position(long newPosition) throws IOException {
     cancelCurrentRequest();
     invalidateBufferedContent();
 
+    if (footerBuffer == null && newPosition > 0 && newPosition >= footerStartOffsetInBytes) {
```

Comment:
I think that `position` should not perform any actual IO, because some clients can call it multiple times until they will actually call read.

Reply:
Just curious. Why would a client call position (BTW this is `seek`) multiple times before reading?

I agree that semantically this is not the right place to add, but it keeps the code much cleaner. Let me know if you have a strong view on this. I can change the implementation.

Reply:
moved the prefetch to read